lib/rules/indent.js

```diff
@@ -608,9 +608,13 @@ module.exports = {
                 return elem !== null;
             });
 
-            // Skip if first element is in same line with this node
-            if (elements.length > 0 && elements[0].loc.start.line === node.loc.start.line) {
-                return;
+            // Skip if element is in same line with this node
+            if (elements.length > 0 && node.type === "ArrayExpression") {
+                elements.forEach(el => {
+                    if (el.loc.start.line === node.loc.start.line) {
+                        return;
+                    }
+                });
```

Comment:
The issue is that `return` isn't doing anything here; it's returning from the `forEach` function, but it looks like it was supposed to return from `checkIndentInArrayOrObjectBlock`.